pkgs/desktops/gnome/apps/fragments/default.nix

```diff
@@ -0,0 +1,86 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, substituteAll
+, meson
+, vala
+, ninja
+, pkg-config
+, cmake
+, wrapGAppsHook
+, desktop-file-utils
+, appstream-glib
+, python3
+, glib
+, gtk3
+, libhandy
+, libtransmission
+, libb64
+, libutp
+, miniupnpc
+, dht
+, libnatpmp
+, libevent
+, curl
+, openssl
+, zlib
+}:
+
+stdenv.mkDerivation rec {
+  pname = "fragments";
+  version = "1.5";
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = "Fragments";
+    rev = version;
+    sha256 = "0x1kafhlgyi65l4w67c24r8mpvasg3q3c4wlgnjc9sxvp6ki7xbn";
+  };
+
+  patches = [
+    # Fix dependency resolution
+    (substituteAll {
+      src = ./dependency-resolution.patch;
+      curl = lib.getLib curl;
+      openssl = lib.getLib openssl;
+      zlib = lib.getLib zlib;
+    })
+  ];
+
+  nativeBuildInputs = [
+    meson
+    vala
+    ninja
+    pkg-config
+    cmake
```

Comment:
Are you sure we need all 4 build tools?

Reply:
Yes, it's required to satisfy the configure script

Reply:
The cmake requirement turned out to just be a warning, but all the other ones are hard dependencies